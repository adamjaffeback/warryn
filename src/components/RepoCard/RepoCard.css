.RepoCard {
  background-color: #f2f2f2;
  border-radius: 2px;
  margin: 1rem;
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
  width: 30vw;
  height: 10vh;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  cursor: pointer;
}

.RepoCard.disabled {
  background-color: grey;
  cursor: not-allowed;
}

.RepoCard.selected {
  background-color: yellow;
}
