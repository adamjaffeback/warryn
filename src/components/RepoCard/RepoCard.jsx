import React from 'react';
import './RepoCard.css';

function RepoCard({repo, clickCard}) {
  const isDisabled = repo.open_issues ? '' : 'disabled';
  const isSelected = repo.selected ? 'selected' : '';
  const classes = `RepoCard ${isDisabled} ${isSelected}`;

  function handleClick () {
    if (repo.open_issues > 0) {
      clickCard(repo.id);
    }
  }

  return (
    <div className={classes} onClick={handleClick}>
      <div className='RepoCard-title'>{repo.name}</div>
      <div className='RepoCard-issue-count'>{repo.open_issues} open issues</div>
    </div>
  );
}

export default RepoCard;
