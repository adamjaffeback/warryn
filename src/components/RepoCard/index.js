import React from 'react';
import './RepoCard.css';

function RepoCard({repo, clickCard}) {
  const isDisabled = repo.open_issues ? '' : 'disabled';
  const isSelected = repo.selected ? 'selected' : '';
  const classes = `RepoCard ${isDisabled} ${isSelected}`;

  function handleClick () {
    if (repo.open_issues > 0) {
      clickCard(repo.id);
    }
  }

  return (
    <div className={classes} onClick={handleClick}>
      <div>{repo.name}</div>
      <div>Open Issues: {repo.open_issues}</div>
    </div>
  );
}

export default RepoCard;
