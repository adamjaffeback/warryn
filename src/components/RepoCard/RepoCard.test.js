import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RepoCard from './RepoCard';

const repo = {
  id: 1,
  name: 'warryn',
  open_issues: 1,
  selected: false,
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RepoCard repo={repo} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should enable the card if the repo has open issues', () => {
  const enabledCard = renderer
   .create(<RepoCard repo={repo} />)
   .toJSON();
 expect(enabledCard).toMatchSnapshot();
});

it('should disable the card if the repo has no open issues', () => {
  const disabledCard = renderer
   .create(<RepoCard repo={{...repo, open_issues: 0}} />)
   .toJSON();
 expect(disabledCard).toMatchSnapshot();
});

it('should not apply a selected class', () => {
  const unselectedCard = renderer
   .create(<RepoCard repo={repo} />)
   .toJSON();
 expect(unselectedCard).toMatchSnapshot();
});

it('should apply a selected class', () => {
  const selectedCard = renderer
   .create(<RepoCard repo={{...repo, selected: true}} />)
   .toJSON();
 expect(selectedCard).toMatchSnapshot();
});
