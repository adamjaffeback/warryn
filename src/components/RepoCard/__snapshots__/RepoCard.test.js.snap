// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should apply a selected class 1`] = `
<div
  className="RepoCard  selected"
  onClick={[Function]}
>
  <div
    className="RepoCard-title"
  >
    warryn
  </div>
  <div
    className="RepoCard-issue-count"
  >
    1
     open issues
  </div>
</div>
`;

exports[`should disable the card if the repo has no open issues 1`] = `
<div
  className="RepoCard disabled "
  onClick={[Function]}
>
  <div
    className="RepoCard-title"
  >
    warryn
  </div>
  <div
    className="RepoCard-issue-count"
  >
    0
     open issues
  </div>
</div>
`;

exports[`should enable the card if the repo has open issues 1`] = `
<div
  className="RepoCard  "
  onClick={[Function]}
>
  <div
    className="RepoCard-title"
  >
    warryn
  </div>
  <div
    className="RepoCard-issue-count"
  >
    1
     open issues
  </div>
</div>
`;

exports[`should not apply a selected class 1`] = `
<div
  className="RepoCard  "
  onClick={[Function]}
>
  <div
    className="RepoCard-title"
  >
    warryn
  </div>
  <div
    className="RepoCard-issue-count"
  >
    1
     open issues
  </div>
</div>
`;
